src/Tinker.jl

```diff
@@ -14,13 +14,210 @@ end
+Base.isempty(H::Handle) = isempty(H.r)
+
+# Creates handle given a Rectangle and a position
+function Handle(r::Rectangle, pos::String)
+    # Position of handle refers to center coordinate of handle based on rect
+    position_coord = Dict("tlc"=>(r.x,r.y),"ts"=>(r.x+(r.w/2),r.y),
+                          "trc"=>(r.x+r.w,r.y),"rs"=>(r.x+r.w,r.y+(r.h/2)),
+                          "brc"=>(r.x+r.w,r.y+r.h),"bs"=>(r.x+(r.w/2),r.y+r.h),
+                          "blc"=>(r.x,r.y+r.h),"ls"=>(r.x,r.y+(r.h/2)))
+    if pos=="tlc" || pos=="ts" || pos=="trc" || pos=="rs" || pos=="brc" ||
+       pos=="bs" || pos=="blc" || pos=="ls"
+        x = position_coord[pos][1]
+        y = position_coord[pos][2]
+        return Handle(r,pos,x,y)
+    else
+        println("Not a valid Handle position.")
+        return Handle()
+    end
+end
+
+# Draws a handle
+function drawhandle(ctx, handle::Handle, color)
+    if !isempty(handle)
+        set_source(ctx,color)
+        d = 8 # physical dimension of handle
+        rectangle(ctx, handle.x-(d/2), handle.y-(d/2),
+                  d, d)
+        stroke(ctx)
+    end
+end; # like drawrect, but makes x,y refer to center of handle
+
+# Returns true if a handle is clicked
+function is_clicked(pt::XY, handle::Handle)
+    if (handle.x - 5 < pt.x < handle.x + 5) &&
+        (handle.y - 5 < pt.y < handle.y + 5)
+        return true
+    else
+        return false
+    end
+end
+
+# A rectangle with handles at all 8 positions
+struct RectHandle
+    r::Rectangle
+    h::NTuple{8,Handle}
+end
+
+RectHandle() = RectHandle(Rectangle())
+Base.isempty(RH::RectHandle) = isempty(RH.r)
+
+# Creates a RectHandle given just a Rectangle
+function RectHandle(r::Rectangle)
+    # derive all 8 handles from r
+    # numbered 1-8, 1=tlc, moving clockwise around rectangle
+    h = (Handle(r, "tlc"), Handle(r, "ts"), Handle(r, "trc"), Handle(r, "rs"),
+         Handle(r, "brc"), Handle(r, "bs"), Handle(r, "blc"), Handle(r, "ls"))
+    return RectHandle(r,h)
+end
+
+# Draws RectHandle
+function drawrecthandle(ctx, rh::RectHandle, color1, color2)
+    drawrect(ctx, rh.r, color1)
+    for n in 1:length(rh.h)
+        drawhandle(ctx, rh.h[n], color2)
+    end
+end
+
+# Given a RectHandle and a position, returns the corresponding Handle
+function get_handle(rh::RectHandle, pos::String)
+    if pos == rh.h[1].pos
+        return rh.h[1]
+    elseif pos == rh.h[2].pos
+        return rh.h[2]
+    elseif pos == rh.h[3].pos
+        return rh.h[3]
+    elseif pos == rh.h[4].pos
+        return rh.h[4]
+    elseif pos == rh.h[5].pos
+        return rh.h[5]
+    elseif pos == rh.h[6].pos
+        return rh.h[6]
+    elseif pos == rh.h[7].pos
+        return rh.h[7]
+    elseif pos == rh.h[8].pos
+        return rh.h[8]
+    else
+        println("Not a valid position")
+        return Handle()
+    end
+end
+
+# Gets anchor point for modification by handle
+function get_p1(h::Handle, rh::RectHandle)
+    if h.pos[end] == 'c' # 'h' is a corner
```

Comment:
I wasn't sure at first, but I really like the string encoding. :+1: 

This one is less of a big deal, but you might simplify some of this with
```julia
const opposite_dict = Dict('b'=>'t', 't'=>'b', 'l'=>'r', 'r'=>'l')
```